classes/ActionScheduler_ListTable.php

```diff
@@ -0,0 +1,385 @@
+		}
+	}
+
+	/**
+	 * Implements the logic behind running an action. PP_Table_List validates the request and their
+	 * parameters are valid.
+	 */
+	protected function row_action_run( $action_id ) {
+		try {
+			ActionScheduler::runner()->process_action( $action_id );
+			$success = 1;
+		} catch ( Exception $e ) {
+			$success = 0;
+		}
+
+		set_transient( 'actionscheduler_admin_executed', compact( 'action_id', 'success' ), 30 );
+	}
+
+	/**
+	 * {@inheritDoc}
+	 */
+	public function prepare_items() {
+		$this->process_bulk_action();
+
+		$this->process_row_actions();
+
+		if ( ! empty( $_REQUEST['_wp_http_referer'] ) ) {
+			// _wp_http_referer is used only on bulk actions, we remove it to keep the $_GET shorter
+			wp_redirect( remove_query_arg( array( '_wp_http_referer', '_wpnonce' ), wp_unslash( $_SERVER['REQUEST_URI'] ) ) );
+			exit;
+		}
+
+		$this->prepare_column_headers();
+
+		$per_page = $this->get_items_query_limit();
+		$query = array(
+			'per_page' => $per_page,
+			'offset'   => $this->get_items_query_offset(),
+			'status'   => $this->get_request_status(),
+			'orderby'  => 'modified',
+			'order'    => 'ASC',
+		);
+
+		$this->items = array();
+
+		$total_items = $this->stores->action->query_actions_count( $query );
+
+		foreach ( $this->stores->action->query_actions( $query ) as $id ) {
+			$item = $this->stores->action->fetch_action( $id );
+			$this->items[ $id ] = array(
+				'ID'     => $id,
+				'hook'   => $item->get_hook(),
+				'status' => $this->get_status( $item ),
+				'args'   => $item->get_args(),
+				'group'  => $item->get_group(),
+				'recurrence' => $this->get_recurrence( $item ),
+				'scheduled'  => $item->get_schedule(),
+				'claim_id'   => '',
+				'comments'   => $this->stores->log->get_logs( $id ),
+			);
+		}
+
+		$this->set_pagination_args( array(
+			'total_items' => $total_items,
+			'per_page'    => $per_page,
+			'total_pages' => ceil( $total_items / $per_page ),
+		) );
+	}
+
+	/**
+	 * This function is used to filter the actions by a status. It will return the status based on a 
+	 * a GET parameter or it will filter by 'pending' by default. 
+	 * We filter by status because if we do not filter we have NullActions that are not rendered at the 
+	 * moment.
+	 *
+	 * @return string
+	 */
+	public function get_request_status() {
+		$statuses = array(
+			ActionScheduler_Store::STATUS_PENDING,
+			ActionScheduler_Store::STATUS_COMPLETE,
+			ActionScheduler_Store::STATUS_FAILED,
+		);
+
+		if ( ! empty( $_GET['status'] ) && in_array( $_GET['status'], $statuses ) ) {
+			return $_GET['status'];
+		}
+
+		return ActionScheduler_Store::STATUS_PENDING;
+	}
+
+	/**
+	 * Prints the available statuses so the user can click to filter.
+	 *
+	 * @return void
+	 */
+	public function display_filter_by_status() {
+		$statuses = array(
+			'pending' => ActionScheduler_Store::STATUS_PENDING,
+			'complete' => ActionScheduler_Store::STATUS_COMPLETE,
+			'failed'  => ActionScheduler_Store::STATUS_FAILED,
```

Comment:
What about `ActionScheduler_Store::STATUS_CANCELED` and `ActionScheduler_Store::STATUS_RUNNING`?